unstructured/ingest/connector/weaviate.py

```diff
@@ -0,0 +1,167 @@
+import json
+import typing as t
+from dataclasses import dataclass
+
+from unstructured.ingest.error import DestinationConnectionError, SourceConnectionError
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class SimpleWeaviateConfig(BaseConnectorConfig):
+    host_url: str
+    class_name: str
+    auth_keys: t.Optional[t.Dict[str, str]] = None
+
+
+@dataclass
+class WeaviateWriteConfig(WriteConfig):
+    batch_size: int = 100
+
+
+@dataclass
+class WeaviateDestinationConnector(BaseDestinationConnector):
+    write_config: WeaviateWriteConfig
+    connector_config: SimpleWeaviateConfig
+
+    @requires_dependencies(["weaviate"], extras="weaviate")
+    @DestinationConnectionError.wrap
+    def initialize(self):
+        from weaviate import Client
+
+        auth = self._resolve_auth_method()
+        self.client: Client = Client(url=self.connector_config.host_url, auth_client_secret=auth)
+
+    @requires_dependencies(["weaviate"], extras="weaviate")
+    def check_connection(self):
+        from weaviate import Client
+
+        try:
+            auth = self._resolve_auth_method()
+            _ = Client(url=self.connector_config.host_url, auth_client_secret=auth)
```

Comment:
Can we leverage this pattern to share a client once it's been created since it's potentially being created more than once if the check connection method is called after the initialize one, example: [mongodb.py#L62](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/connector/mongodb.py#L62)